scarb/tests/package.rs

```diff
@@ -1525,3 +1525,68 @@ fn package_with_publish_disabled() {
             [..]Packaged [..] files, [..] ([..] compressed)
         "#});
 }
+
+#[test]
+fn package_with_package_script() {
+    let t = TempDir::new().unwrap();
+
+    #[cfg(not(windows))]
+    let script_name = "script.sh";
+    #[cfg(not(windows))]
+    let script_code = indoc! { r#"
+      touch text.txt
+      cd ../..
+      cargo build
+    "#};
+
+    #[cfg(windows)]
+    let script_name = "script.bat";
+    #[cfg(windows)]
+    let script_code = indoc! { r#"
+      @echo off
+      copy NUL text.txt
+      cd ../..
+      cargo build
+    "#};
+
+    CairoPluginProjectBuilder::start()
+        .name("foo")
+        .scarb_project(|b| {
+            b.name("foo")
+                .version("1.0.0")
+                .manifest_extra(formatdoc! {r#"
+                  [cairo-plugin]
+
+                  [scripts]
+                  package = "{script_name}"
+                "#})
+        })
+        .lib_rs(indoc! {r#"
+          use cairo_lang_macro::{ProcMacroResult, TokenStream, attribute_macro};
+
+          #[attribute_macro]
+          pub fn some(_attr: TokenStream, token_stream: TokenStream) -> ProcMacroResult {
+              ProcMacroResult::new(token_stream)
+          }
+        "#})
+        .src(script_name, script_code)
+        .build(&t);
+
+    #[cfg(not(windows))]
+    make_executable(&t.to_path_buf().join(script_name));
+
+    Scarb::quick_snapbox()
+        .arg("package")
+        .current_dir(&t)
+        .assert()
+        .success();
+
+    assert!(Path::new(
+        &t.to_path_buf()
+            .join("target")
+            .join("package")
+            .join("text.txt"),
```

Comment:
Maybe we should check whether something gets produces in `target/scarb/cairo-plugin'? 

I've tried running this locally and it doesn't seem to produce anything, possibly due to `prepare_archive_recipe` not copying contents of `target`.

I've also tried changing `.gitignore` to exclude `target/scarb/cairo-plugin`, but it didn't seem to help. This is probably also something we should address:
1. The main purpose of the package script is to produce binaries in `target/scarb/cairo-plugin`. We might want to include warning on this directory being empty
2. I think we should include `target/scarb/cairo-plugin` files regardless of whether they are .gitignore'd, as we probably don't want to force users to include binaries in their repos.

Reply:
Previous version didn't have the packaging the result dlls implemented yet, but right now it's done